mapeditor/mapview.cpp

```diff
@@ -0,0 +1,448 @@
+#include "StdInc.h"
+#include "mapview.h"
+#include "mainwindow.h"
+#include <QGraphicsSceneMouseEvent>
+#include "mapcontroller.h"
+
+MinimapView::MinimapView(QWidget * parent):
+	QGraphicsView(parent)
+{
+	// Disable scrollbars
+	setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+	setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+}
+
+void MinimapView::dimensions()
+{
+	fitInView(0, 0, controller->map()->width, controller->map()->height, Qt::KeepAspectRatio);
+}
+
+void MinimapView::setController(MapController * ctrl)
+{
+	controller = ctrl;
+}
+
+void MinimapView::mouseMoveEvent(QMouseEvent *mouseEvent)
+{
+	this->update();
+	
+	auto * sc = static_cast<MinimapScene*>(scene());
```

Comment:
when sc can be nullptr? When there is no scene()?

Reply:
yes, on application start